src/aiida_quantumespresso/parsers/projwfc.py

```diff
@@ -2,458 +2,423 @@
+
+        # Parse the projection arrays
+        energy_pattern = re.compile(r'\n====.+==== \n')
+        psi_pattern = re.compile(r'([.\d]+)\*\[#\s*(\d+)\]')
+
+        projections = []
+
+        for block in kpoint_blocks:
+
+            kpoint_projections = []
+
+            for band_projections in re.split(energy_pattern, block)[1:]:
+
+                proj_array = np.zeros(num_orbitals)
+
+                for [projection_value, orbital_index] in psi_pattern.findall(band_projections):
+                    proj_array[int(orbital_index) - 1] = projection_value
+
+                kpoint_projections.append(proj_array)
+
+            projections.append(kpoint_projections)
+
+        projections = np.array(projections)
+
+        return bands, projections
+
+    def _parse_pdos_files(self, retrieved_temporary_folder: Path, nspin: int,
+                          spinorbit: bool, logs: AttributeDict) -> Tuple[ArrayLike, ArrayLike, ArrayLike]:
+        """Parse the PDOS files and convert them into arrays.
+
+        Reads in all of the ``*.pdos*`` files and converts the data into arrays. The PDOS arrays are then concatenated
+        and reordered so the order of the columns matches the order of the orbitals read from the statelines. This is
+        done in three steps:
+
+        1. Sort the filenames by the atom # and wfc #, in that order.
+        2. Concatenate all the PDOS files into one large array.
+        3. Reorder the columns depending on the ``npsin`` value and if spin-orbit is used.
+
+        :param retrieved_temporary_folder: temporary folder of retrieved files that is deleted after parsing.
+        :param nspin: nspin value of the parent calculation.
+        :param spinorbit: True if the calculation used spin-orbit coupling.
+
+        :return: tuple of three arrays containing the energy grid, the total DOS and the PDOS
+        """
+
+        def natural_sort_key(sort_key, _nsre=re.compile('([0-9]+)')):
+            """Pass to ``key`` for ``str.sort`` to achieve natural sorting. For example, ``["2", "11", "1"]`` will be sorted to
+            ``["1", "2", "11"]`` instead of ``["1", "11", "2"]``
+
+            :param sort_key: Original key to be processed
+            :return: A list of string and integers.
+            """
+            keys = []
+            for text in _nsre.split(sort_key):
+                if text.isdigit():
+                    keys.append(int(text))
+                else:
+                    keys.append(text)
+            return keys
+
+        # Read the `pdos_tot` file
+        try:
+            pdostot_filepath = next(retrieved_temporary_folder.glob('*pdos_tot*'))
+            with pdostot_filepath.open('r') as pdostot_file:
+                # Columns: Energy(eV), Ldos, Pdos
+                pdostot_array = np.atleast_2d(np.genfromtxt(pdostot_file))
+                energy = pdostot_array[:, 0]
+                dos = pdostot_array[:, 1]
```

Comment:
In a spin-polarized calculation (`nspin ==2 `, we only parse the spin-up column at the moment). The suggestion is following the logic of the `dos.x` Parser: 

https://github.com/aiidateam/aiida-quantumespresso/blob/c84ef21dbea1109814e9780d1d18e7303fa2fc25/src/aiida_quantumespresso/parsers/dos.py#L51-L69

```suggestion
                pdostot_array = np.atleast_2d(np.genfromtxt(pdostot_file))
                
                dos = []
                dos_names = []
                energy = pdostot_array[:, 0]
                if nspin ==2:
                    dos += [pdostot_array[:, 1]]
                    dos_names += ['dos_spin_up']
                    dos += [pdostot_array[:, 2]]
                    dos_names += ['dos_spin_down']
                else:
                    dos += [pdostot_array[:, 1]]
                    dos_names += ['dos']
```